tests/fsharp/nunitConf.fs

```diff
@@ -0,0 +1,379 @@
+        |> mapWithDefaults ( [ "FSCBINPATH", FSCBinPath ] |> Map.ofList )
+
+    let configurationName =
+        match env |> Map.tryFind "FSHARP_TEST_SUITE_CONFIGURATION" |> Option.map parseConfigurationName with
+        | Some confName -> confName
+        | None -> configurationName
+
+    processor {
+        do! updateCmd env { Configuration = configurationName; Ngen = doNgen; }
+            |> Attempt.Run
+            |> function Success () -> Success () | Failure msg -> genericError msg ()
+
+        let cfg =
+            let c = config env
+            let usedEnvVars =
+                c.EnvironmentVariables 
+                |> Map.add "FSC" c.FSC             
+            { c with EnvironmentVariables = usedEnvVars }
+
+        logConfig cfg
+
+        let directoryExists = Commands.directoryExists (Path.GetTempPath()) >> Option.isSome 
+
+        let checkfscBinPath () = processor {
+
+            let fscBinPath = cfg.EnvironmentVariables |> Map.tryFind "FSCBINPATH"
+            return!
+                match fscBinPath with
+                | Some dir when directoryExists dir -> Success
+                | None -> genericError "environment variable 'FSCBinPath' is required to be a valid directory, is not set"
+                | Some dir -> genericError (sprintf "environment variable 'FSCBinPath' is required to be a valid directory, but is '%s'" dir)
+            }
+
+        let smokeTest () = processor {
+            let tempFile ext = 
+                let p = Path.ChangeExtension( Path.GetTempFileName(), ext)
+                File.AppendAllText (p, """printfn "ciao"; exit 0""")
+                p
+
+            let tempDir = Commands.createTempDir ()
+            let exec exe args =
+                log "%s %s" exe args
+                use toLog = redirectToLog ()
+                Process.exec { RedirectError = Some toLog.Post; RedirectOutput = Some toLog.Post; RedirectInput = None } tempDir cfg.EnvironmentVariables exe args
+
+            do! Commands.fsc exec cfg.FSC "" [ tempFile ".fs" ] |> checkResult
+
+            do! Commands.fsi exec cfg.FSI "" [ tempFile ".fsx" ] |> checkResult
+        
+            }
+    
+        do! checkfscBinPath ()
+
+        do! smokeTest ()
+
+        return cfg
+    } 
+
+
+let suiteHelpers = lazy (
+    initializeSuite ()
+    |> Attempt.Run 
+    |> function Success x -> x | Failure err -> failwith (sprintf "Error %A" err)
+)
+
+[<AttributeUsage(AttributeTargets.Assembly)>]
+type public InitializeSuiteAttribute () =
+    inherit TestActionAttribute()
+
+    override x.BeforeTest details =
+        if details.IsSuite 
+        then suiteHelpers.Force() |> ignore
+
+    override x.AfterTest details =
+        ()
+
+    override x.Targets with get() = ActionTargets.Test ||| ActionTargets.Suite
+
+
+[<assembly:InitializeSuite()>]
+()
+
+module FSharpTestSuite =
+
+    let getTagsOfFile path =
+        match File.ReadLines(path) |> Seq.take 5 |> Seq.tryFind (fun s -> s.StartsWith("// #")) with
```

Comment:
I want to read the header of the file but is not always the same ( shebang, white lines etc)
I'll add a comment, is a simplified parsing implementation